### PR TITLE
Semantics framework updates

### LINKED ISSUE #3815: Add support for a semantic escape
iOS accessibility has the notion of an escape used to exit a modal or hierarchical view.  Users can trigger an escape quickly with a VoiceOver action.  We should add support for this.

### LINKED ISSUE #11140: a11y: Persistent bottom sheet cannot be dismissed on iOS

### LINKED ISSUE #14468: Image widget should have semanticsLabel property like Icon

### PR DESCRIPTION
**Pull request by jonahwilliams**
Requires https://github.com/flutter/engine/pull/5601

Fixes https://github.com/flutter/flutter/issues/14468
Fixes https://github.com/flutter/flutter/issues/3815
Fixes https://github.com/flutter/flutter/issues/11140

Partial fix of https://github.com/flutter/flutter/issues/16673
Android fix of https://github.com/flutter/flutter/issues/16395
Android fix of https://github.com/flutter/flutter/issues/14378

Changes:
  * Move the SemanticsConfiguration update from RenderToggleable to each subclass, so that Switch can use toggled.
  * Add image, liveRegion, toggled properties to Semantics, SemanticsConfiguration, SemanticsNode
  * Added semanticsLabel and excludeFromSemantics to Image (the latter so that we avoid creating a semantics node)
  * Added onDismiss semantics action which maps to the modal escape on iOS and dismiss action on Android.
  * Added dismiss and liveRegion to snackbar widget
  * Updated custom painter semantics to handle image, liveRegion, toggle
  * Updated relevant tests to use correct flag/action

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -553,6 +566,33 @@ class SemanticsProperties extends DiagnosticableTree {
...
   final bool namesRoute;
 
+  /// If non-null, whether the node represents an image.
+  ///
+  /// See also:
+  ///
+  ///   * [SemanticsFlag.image], for the flag this setting controls.
+  final bool image;
+
+  /// If non-null, whether the node should be considered a live region.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart]: Great doc comment!

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 2]
```diff
@@ -693,7 +733,7 @@ class SemanticsProperties extends DiagnosticableTree {
   /// vertically scrollable.
   ///
   /// VoiceOver users on iOS can trigger this action by swiping down with three
-  /// fingers. TalkBack users on Android can trigger this action by swiping
+  /// fingers. TalkBack users on  Android can trigger this action by swiping
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart]: remove extra white space?

File: packages/flutter/lib/src/semantics/semantics_event.dart [hunk 3]
```diff
@@ -133,3 +133,21 @@ class TapSemanticEvent extends SemanticsEvent {
   @override
   Map<String, dynamic> getDataMap() => const <String, dynamic>{};
 }
+
+/// An event which triggers an announcement of a live region.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics_event.dart]: ... a polite announcement ... ?

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 4]
```diff
@@ -553,6 +566,33 @@ class SemanticsProperties extends DiagnosticableTree {
...
+  /// If non-null, whether the node represents an image.
+  ///
+  /// See also:
+  ///
+  ///   * [SemanticsFlag.image], for the flag this setting controls.
+  final bool image;
+
+  /// If non-null, whether the node should be considered a live region.
+  ///
+  /// On Android, when a live region semantics node is first created TalkBack
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart]: Can you document what iOS does when this is set?

File: packages/flutter/lib/src/semantics/semantics_event.dart [hunk 5]
```diff
@@ -133,3 +133,21 @@ class TapSemanticEvent extends SemanticsEvent {
...
+///
+/// This requires that the semantics node has already been marked as a live
+/// region. Only Android (TalkBack) will make a verbal announcement, as long as
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics_event.dart]: What does iOS do?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics_event.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics.dart]: 🌮

File: packages/flutter/lib/src/semantics/semantics_event.dart (hunk 5 above)
Comment (jonahwilliams) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: Nothing :(

Comment (goderbauer) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: I would probably spell that out in the doc comment.

File: packages/flutter/lib/src/semantics/semantics_event.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: Looks like it didn't come through?

File: packages/flutter/lib/src/semantics/semantics_event.dart (hunk 5 above)
Comment (goderbauer) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: Never mind. I see it's there now. Was looking at an old diff :(

File: packages/flutter/lib/src/semantics/semantics_event.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter/lib/src/semantics/semantics_event.dart]: Never mind. I see it's there now. Was looking at an old diff :(

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 6]
```diff
@@ -3136,6 +3137,9 @@ class RenderSemanticsAnnotations extends RenderProxyBox {
     bool scopesRoute,
     bool namesRoute,
     bool hidden,
+    bool image,
+    bool liveRegion,
+    bool isSwitch,
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: this argument is ignored. was that a mistake?

Comment (jonahwilliams) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: yup

Comment (Hixie) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: i have a PR out to remove it

**Pull request merged**